### PR TITLE
Containerizing oAuth app

### PR DESCRIPTION
**Pull request by schaitanya**
- Adding Dockerfile
- Fixing package.json
- Adding yarn.lock file

@vnguyener please check if this will change any existing packages / providers

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+RUN groupadd -r kickbox && \
+    useradd -r -g kickbox kickbox && \
+    chown -R kickbox:kickbox /srv/oauth
+
+# Install CoffeeScript
+RUN yarn global add coffeescript@1.10.0
+USER kickbox
+COPY --chown=kickbox --from=build /tmp /srv/oauth
+COPY --chown=kickbox . /srv/oauth/
+RUN npm run grunt
```
Comment (rathboma) [inline Dockerfile]: why do we need grunt?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+RUN yarn --pure-lockfile
+
+FROM node:6-slim
+WORKDIR /srv/oauth
+RUN groupadd -r kickbox && \
+    useradd -r -g kickbox kickbox && \
+    chown -R kickbox:kickbox /srv/oauth
+
+# Install CoffeeScript
+RUN yarn global add coffeescript@1.10.0
```
Comment (rathboma) [inline Dockerfile]: can this just be in the package.json?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+USER kickbox
+COPY --chown=kickbox --from=build /tmp /srv/oauth
+COPY --chown=kickbox . /srv/oauth/
+RUN npm run grunt
+
+CMD ["./node_modules/.bin/coffee", "/srv/oauth/lib/oauthd.coffee"]
```
Comment (rathboma) [inline Dockerfile]: lets make this a package command so we can do `yarn run server`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+FROM node:6 as build
+WORKDIR /tmp
+COPY package.json yarn.lock /tmp/
+RUN yarn --pure-lockfile
```
[the remark this answers is not in the document]
Comment (rathboma) [reply Dockerfile]: see https://yarnpkg.com/en/docs/cli/install#toc-yarn-install-frozen-lockfile

File: Dockerfile (hunk 2 above)
Comment (schaitanya) [reply Dockerfile]: Installing global packages. But I'll remove this in favor of "yarn run".

**Comment by schaitanya** [comment]
@rathboma updated the Dockerfile

**Pull request merged**